app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -387,45 +376,97 @@ class BrowserTabViewModel(
         browserViewState.value = currentState.copy(isFullScreen = false)
     }
 
-    override fun loadingStarted(url: String?) {
-        Timber.v("Loading started")
-        if (!currentBrowserViewState().browserShowing) return
-        val progress = currentLoadingViewState()
-        loadingViewState.value = progress.copy(isLoading = true)
-        pendingUrl = url
-        site = null
-        onSiteChanged()
-    }
+    override fun navigationStateChanged(newWebNavigationState: WebNavigationState) {
 
-    override fun navigationOptionsChanged(navigation: BrowserNavigationOptions) {
-        navigationOptions = navigation
+        val stateChange = newWebNavigationState.compare(webNavigationState)
+        webNavigationState = newWebNavigationState
 
         if (!currentBrowserViewState().browserShowing) return
 
         browserViewState.value = currentBrowserViewState().copy(
-            canGoBack = navigation.canGoBack || !skipHome,
-            canGoForward = navigation.canGoForward
+            canGoBack = newWebNavigationState.canGoBack || !skipHome,
+            canGoForward = newWebNavigationState.canGoForward
         )
+
+        when {
```

Comment:
bump `stateChange` up into `when` clause? e.g., 

    when (stateChange) {
        is NewPage -> pageChanged(stateChange.url, stateChange.title)
        is PageCleared -> pageCleared()
        is UrlUpdated -> urlUpdated(stateChange.url)
    }